### PR TITLE
support otel.instrumentation.common.default-enabled in spring starter

### PR DESCRIPTION
**Pull request by zeitlinger**

### CODE REVIEW

**Review by jeanbisutti — COMMENTED** [review]
The behavior seems different from that of the Java agent: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/4597d716f43c7d882632f76575baadcc1103486a/javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/config/AgentConfig.java#L14

**Comment by zeitlinger** [comment]
> The behavior seems different from that of the Java agent:
> 
> https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/4597d716f43c7d882632f76575baadcc1103486a/javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/config/AgentConfig.java#L14

The comment is wrong, here's how it should be:

```java
  public static boolean isInstrumentationEnabled(
      ConfigProperties config, Iterable<String> instrumentationNames, boolean defaultEnabled) {
    // If default is enabled, we want to disable individually,
    // if default is disabled, we want to enable individually.
    boolean anyEnabled = defaultEnabled;
    for (String name : instrumentationNames) {
      String propertyName = "otel.instrumentation." + name + ".enabled";
      boolean enabled = config.getBoolean(propertyName, defaultEnabled);

      if (defaultEnabled) {
        anyEnabled &= enabled;
      } else {
        anyEnabled |= enabled;
      }
    }
    return anyEnabled;
  }
```

I think we have the same logic, but it can't hurt to just re-use the method

**Comment by zeitlinger** [comment]
@jeanbisutti I tried it out - not sure it it's worth it: https://github.com/zeitlinger/opentelemetry-java-instrumentation/compare/spring-commo-disabled...zeitlinger:opentelemetry-java-instrumentation:reuse-agent-config?expand=1

File: instrumentation/spring/spring-boot-autoconfigure/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/web/SpringWebInstrumentationAutoConfigurationTest.java [hunk 1]
```diff
@@ -41,6 +41,7 @@ class SpringWebInstrumentationAutoConfigurationTest {
   void instrumentationEnabled() {
     contextRunner
         .withPropertyValues("otel.instrumentation.spring-web.enabled=true")
+        .withPropertyValues("otel.instrumentation.common.default-enabled=false")
```
Comment (jeanbisutti) [inline instrumentation/spring/spring-boot-autoconfigure/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/web/SpringWebInstrumentationAutoConfigurationTest.java:44]: Could you please add a test for 

```
        .withPropertyValues("otel.instrumentation.spring-web.enabled=false")
        .withPropertyValues("otel.instrumentation.common.default-enabled=true")
```

**Review by jeanbisutti — APPROVED** [review]

**Pull request merged**